src/sync.cpp

```diff
@@ -97,27 +97,29 @@ static void potential_deadlock_detected(const LockPair& mismatch, const LockStac
     LogPrintf("POTENTIAL DEADLOCK DETECTED\n");
     LogPrintf("Previous lock order was:\n");
     for (const LockStackItem& i : s1) {
+        std::string_view prefix{};
```

Comment:
Is there any compelling reason to use the "potentially dangerous" `std::string_view` over `std::string` here? (Not claiming that this the use here is dangerous as currently written.)

Lifetime issues are scary and personally I find it _much_ easier to rule out lifetime issues when reviewing code that uses `std::string` compared to code that uses `std::string_view` :)

Somewhat related: [std::string_view encourages use-after-free](https://github.com/isocpp/CppCoreGuidelines/issues/1038)

Reply:
string_view is no different than span, which is also in our code base. I personally prefer string_view over c-strings (raw pointers).

Luckily, if review doesn't catch use-after-free, we have compilers that check the `lifetimebound` attribute (clang), as well as runtime sanitizers.

Reply:
Also, this is only used in tests and the program will `std::abort` anyway right after, so I claim it wouldn't matter if a use-after-free made it in (despite all QA).

Reply:
Force pushed `std::string` to avoid the discussion :sweat_smile: